net/tcp/tcp.h

```diff
@@ -104,6 +104,7 @@
 /* The TCP options flags */
 
 #define TCP_WSCALE            0x01U /* Window Scale option enabled */
+#define TCP_PREALLOC          0x02U /* TCP connection is preallocated */
```

Comment:
Do not use the flag of tcp option field, if the alloc policy will be applied to other protocols,  use [(struct socket_conn_s)](https://github.com/apache/nuttx/blob/master/include/nuttx/net/net.h#L192-L226)

Reply:
Fixed. Used alternative approach instead.